### PR TITLE
Implemented double Schechter function for red galaxies.

### PR DESCRIPTION
**Pull request by nkhadka21**
This PR implements double Schechter function for red galaxies. Also, compares slsim galaxy distribution with DC2 and Diffsky galaxy distribution.

### CODE REVIEW

File: notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb [hunk 1]
```diff
@@ -0,0 +1,758 @@
+{
```
Comment (sibirrer) [inline notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #85.    plt.show()
</pre><p>can you cut these plots at mag=30 to better see what's in the more relevant part of the magnitude range. Can you also make a plot where you show the evolution as a function of redshift in one plot? By eye, I can't even see an evolution in the luminosity function as a function of redshift for the SkyPy distributions, which makes me worried</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/LSST-strong-lensing/slsim/pull/157/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (sibirrer) [inline notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb:1]: <p>I seem to see still a step-function like behavior in the SLSim distributions. Can we make the redshift interval smaller?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/LSST-strong-lensing/slsim/pull/157/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

File: data/SkyPy/lsst-like_new.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+# These are the luminosity function parameters according to M = a_m * z + b_m
+# and Phi = b_phi * exp(a_phi * z). As in Lopez-Sanjuan the model is given as
+# M = M0 + Q * (z - 0.5) and log10(Phi) = Phi0 + P * (z - 0.5) we have to convert
+# the given parameters as a_m = Q, b_m = M0 - 0.5*Q, a_phi = ln(10) * P and
+# b_phi = 10**(Phi0 - 0.5*P)
+cosmology: !astropy.cosmology.default_cosmology.get []
+z_range: !numpy.arange [0.0, 5.01, 0.04]
+M_star: !astropy.modeling.models.Linear1D [-1.03, -20.485]
```
Comment (sibirrer) [inline data/SkyPy/lsst-like_new.yml:8]: are these M_star for the blue or red distribution? Perhaps specify these parameters better

File: notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb (hunk 1 above)
Comment (nkhadka21) [reply notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb:1]: decreased the redshift step to 0.01 which will solve this.

**Comment by ajshajib** [comment]
@nkhadka21, please pull from the main branch to make the coverage report start working for this PR.

File: notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb [hunk 3]
```diff
@@ -0,0 +1,900 @@
...
```
Comment (sibirrer) [inline notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb:1]: <p>is there a possibility to produce these histograms also straight out of SLSim? That would be nice to see what is really been done in SLSim and if things change, then we have checks</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/LSST-strong-lensing/slsim/pull/157/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (nkhadka21) [reply notebooks/validation_notebooks/slsim_dc2_diffsky_galaxy_distribution_comparision.ipynb:1]: Hi @sibirrer , I created this plot directly from SLSim!

**Review by sibirrer — APPROVED** [review]
Looks great, thank you @nkhadka21

**Pull request merged**